Library/Homebrew/cmd/pull.rb

```diff
@@ -190,6 +190,10 @@ def pull
           safe_system 'brew', install, '--debug', f.name
         end
       end
+
+      unless ARGV.include? "--no-repair"
```

Comment:
Instead of the argument let's only run this if we've run it for a tap's formula (check if `tap_name` is set).